tests/sentry/web/frontend/test_auth_organization_login.py

```diff
@@ -997,3 +936,111 @@ def test_correct_redirect_as_2fa_user_invited_member(self):
         )
 
         assert resp.redirect_chain == [("/auth/2fa/", 302)]
+
+
+class OrganizationAuthLoginNoPasswordTest(AuthProviderTestCase):
+    def setUp(self):
+        self.owner = self.create_user()
+        self.organization = self.create_organization(name="foo", owner=self.owner)
+        self.user = self.create_user("bar@example.com", is_managed=False, password="")
+        self.auth_provider = AuthProvider.objects.create(
+            organization=self.organization, provider="dummy"
+        )
+        self.path = reverse("sentry-auth-organization", args=[self.organization.slug])
+        self.auth_sso_path = reverse("sentry-auth-sso")
+        UserEmail.objects.filter(user=self.user, email="bar@example.com").update(is_verified=False)
+
+    def test_flow_authenticated_without_verified_without_password(self):
+        """
+        Given an existing authenticated user, and an updated identity (e.g.
+        the ident changed from the SSO provider), we should be re-linking
+        the identity automatically as they don't have a password.
+
+        This is specifically testing an unauthenticated flow.
+        """
+
+        # setup a 'previous' identity, such as when we migrated Google from
```

Comment:
what is this comment talking about? it doesn't make sense to me in this context

Reply:
i'm not sure, i didn't write this test case, rather moved it to be apart of the new class, i think @RyanSkonnord wrote it so he would know

Reply:
I didn't write it (also originates with https://github.com/getsentry/sentry/commit/1361e383e5ab4b5c1a2782d22efc53ded57602bc) but I believe the intended meaning of "'previous' identity" is the state left behind when a user profile was auto-provisioned in response to an SSO sign-in, and then the corresponding SSO identity was deleted.

In other words, create a `User` entity with no password and imagine that it used to have an associated `AuthIdentity` that now doesn't exist.

Reply:
again, think it was just errant on that original tests, would be in favor of just deleting it here to not cause confusion. i'm also pretty sure the "Google migration" thing is referencing an ancient system that we migrated off of. (see `MigratingIdentityId`)